pkg/client/client_test.go

```diff
@@ -339,6 +339,106 @@ func TestApplySingle(t *testing.T) {
 			assert.Equal(t, "second-value", configMap.Data["first-param"])
 		})
 	})
+
+	t.Run("updates of ServiceAccount", func(t *testing.T) {
+
+		t.Run("should update service account with last applied configuration", func(t *testing.T) {
+			// given
+			// there's an existing SA with secret refs
+			existingSA := newSA()
+			secretRefs := []corev1.ObjectReference{
+				{
+					Name:      "secret",
+					Namespace: existingSA.Namespace,
+				},
+			}
+			existingSA.Secrets = secretRefs
+			existingLastAppliedAnnotation := map[string]string{
+				client.LastAppliedConfigurationAnnotationKey: client.GetNewConfiguration(existingSA),
+			}
+			existingSA.SetAnnotations(existingLastAppliedAnnotation) // let's set the last applied annotation
+			cl, cli := newClient(t)
+			_, err := cl.ApplyRuntimeObject(context.TODO(), existingSA)
```

Comment:
I noticed that we should pass a deep-copy of the object, otherwise, the generation as well as resourceVersion values would be set and would influence the late execution when the object that is passed to the function again: 
```suggestion
			_, err := cl.ApplyRuntimeObject(context.TODO(), existingSA.DeepCopyObject())
```

Reply:
done in https://github.com/codeready-toolchain/toolchain-common/pull/385/commits/91ec6712cad89744600f375013fff20bcbadbe55